### PR TITLE
[SPARK-18530][SS][KAFKA]Change Kafka timestamp column type to TimestampType

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Changed Kafka timestamp column type to TimestampType.

## How was this patch tested?

`test("Kafka column types")`.

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @marmbrus @tdas

File: external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaSourceSuite.scala [hunk 1]
```diff
@@ -413,6 +414,43 @@ class KafkaSourceSuite extends KafkaSourceTest {
...
+
+    val reader = spark
+      .readStream
+      .format("kafka")
+      .option("kafka.bootstrap.servers", testUtils.brokerAddress)
+      .option("kafka.metadata.max.age.ms", "1")
+      .option("startingOffsets", s"earliest")
+      .option("subscribe", topic)
+
+    val kafka = reader.load()
```
Comment (tdas) [inline external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaSourceSuite.scala]: nit: why have separate var for reader

**Comment by tdas** [comment]
I wonder whether it makes sense to actually test with a query that has window and withWatermark on the timestamp column. because that is what we want to enable.

**Comment by tdas** [comment]
LGTM, pending tests.

**Comment by zsxwing** [comment]
@tdas the test already finished. Thanks for reviewing. Merging to master and 2.1.

**Pull request closed without merging**